dlt/destinations/dataset.py

```diff
@@ -74,6 +195,35 @@ def _wrap(*args: Any, **kwargs: Any) -> Any:
 
         return _wrap
 
+    def __copy__(self) -> "ReadableDBAPIRelation":
+        return self.__class__(
+            readable_dataset=self._dataset,
+            provided_query=self._provided_query,
+            table_name=self._table_name,
+            limit=self._limit,
+            selected_columns=self._selected_columns,
+        )
+
+    def limit(self, limit: int) -> "ReadableDBAPIRelation":
+        if self._provided_query:
+            raise ReadableRelationHasQueryException("limit")
+        rel = self.__copy__()
+        rel._limit = limit
+        return rel
+
+    def select(self, selected_columns: List[str]) -> "ReadableDBAPIRelation":
+        if self._provided_query:
+            raise ReadableRelationHasQueryException("select")
+        rel = self.__copy__()
+        rel._selected_columns = selected_columns
+        # NOTE: the line below will ensure that no unknown columns are selected if
+        # schema is known
+        rel.computed_schema_columns
+        return rel
+
+    def head(self) -> "ReadableDBAPIRelation":
+        return self.limit(5)
+
 
 class ReadableDBAPIDataset(SupportsReadableDataset):
```

Comment:
let's move ibis() here. then we hand over the connection and return the ibis backend

Reply:
can we move schema property to the Protocol?